demos/top-app-bar.html

```diff
@@ -172,14 +172,14 @@ <h4 class="mdc-typography--subheading2 mdc-typography--adjust-margin">Common Opt
           </div>
         </div>
         <div class="demo-col">
-          <h4 class="mdc-typography--subheading2 mdc-typography--adjust-margin">Top App Bar Specific Options</h4>
+          <span class="mdc-typography--subheading2 demo-options-heading">Top App Bar Specific Options</span>
           <div>
             <input type="checkbox" id="prominent-checkbox"/>
             <label for="prominent-checkbox">Prominent</label>
           </div>
         </div>
         <div class="demo-col">
-          <h4 class="mdc-typography--subheading2 mdc-typography--adjust-margin">Short Top App Bar Specific Options</h4>
+          <span class="mdc-typography--subheading2 demo-options-heading">Short Top App Bar Specific Options</span>
```

Comment:
This change makes me realize something, related to this PR as well as the task to redefine the typography scales...

If we're going to be introducing typography scales for Headline 1-6, I would assume they're naturally going to often align with `h1` to `h6`. These elements commonly have default margins set by the user agent, which might not be appropriate when combined with our styles. Are we sure we don't want to override those with saner defaults? Or do we think we'd override those in all cases they're used, perhaps, thus obviating the need for adjust-margin anyway?

Reply:
I think typography should only handle font properties (font-family, font-size, letter-spacing, etc) which can be displayed in any HTML element. We should not assume that `@mdc-typography` mixin is applied to a `h3` element.